src/FFmpegWriter.cpp

```diff
@@ -511,17 +525,29 @@ void FFmpegWriter::WriteHeader() {
 	snprintf(oc->AV_FILENAME, sizeof(oc->AV_FILENAME), "%s", path.c_str());
 
 	// Write the stream header, if any
-	// TODO: add avoptions / parameters instead of NULL
 
 	// Add general metadata (if any)
 	for (std::map<string, string>::iterator iter = info.metadata.begin(); iter != info.metadata.end(); ++iter) {
 		av_dict_set(&oc->metadata, iter->first.c_str(), iter->second.c_str(), 0);
 	}
 
-	if (avformat_write_header(oc, NULL) != 0) {
+	// Set multiplexing parameters
+	AVDictionary *dict = NULL;
+
+	bool is_mp4 = strcmp(oc->oformat->name, "mp4");
+	bool is_mov = strcmp(oc->oformat->name, "mov");
+	// Set dictionary preset only for MP4 and MOV files
+	if (is_mp4 || is_mov)
+		av_dict_copy(&dict, mux_dict, 0);
+
+	if (avformat_write_header(oc, &dict) != 0) {
```

Comment:
I didn't analyze the return values in the code, because it is not critical for performance or what ever. If the options cannot be used, I don't mind. The original thought was not to mix my own muxing dictionary with parameters that in future may be written somewhere else.

Reply:
Oh, no, I totally understand about not checking the output dictionary, we don't care about "options that were not found". And I'm not at all concerned about performance, or even memory — I'm sure the AVDictionary is small, and even with the copy they're both getting freed right away. I'm sure it's not an issue.

> The original thought was not to mix my own muxing dictionary with parameters that in future may be written somewhere else

OK, so there may be a need to populate `dict` with other keys besides the ones in `mux_dict`. That makes sense.

My only thought was whether there might be any situation where `FFmpegWriter::WriteHeader()` ends up getting called _again_ later on, after a restart or something... in which case, would there be any use in keeping the `mux_dict` around so those options don't have to be re-set? But I'm not sure that can actually happen, anyway, and even if it does we're probably better off making them call `FFmpegWriter::SetOption() again to re-select the presets... if they still want them applied. Works for me!